### ISSUE TITLE
Min balances

### ISSUE THREAD

**Issue by pjdejoe** [habibitcoin/habibalancer#16]
Hi Team - Function getConfig fails at ctxvalue("configs")

See stacktrace in attached image

Additional question:  Is there a minimum balance that needs to be in strike or Kraken for this to work?

![habibalancer — -zsh — 107×34 2022-11-19 at 1 29 28 PM](https://user-images.githubusercontent.com/114040693/202870553-ef97ff23-fa23-46f0-ba4a-5173cc599593.jpg)

**Comment by habibitcoin** [comment]
Hmm, no you dont need a minimum balance, but this would seem to indicate a config value is empty. Are all values at least set?

Feel free to post your .env file with API keys redacted. Also feel free to reach out to me over Discord: bitasstds#2461

**Comment by JackC27** [comment]
All values are set, except for the optional macaroon field.

Looper continues to execute getConfig which, only once, correctly returns a value based on what we see in the logs, but continues to run even after it returns what seems to be a useful value. 

I think that even though getconfig works once the looper continues to run and overwrites the config variable when it is called by the kraken lightning sendPayReq fn, in step 4 " if krakenEnabled == 'true' " conditional. 
GetConfig differs from LoadConfig in a few ways that may be obvious to your developers but I can't intuitively tell what may be,  if anything, is wrong there as I have zero experience with GO except for the past few days. Maybe ctx within is being overwritten improperly? Maybe the value passed to the argument is passed by reference instead of value by the overwrite is referencing a nil value? 

While troubleshooting GetConfig, we added to each GetConfig functino, call a log of ctx, only once does it return what seems to be a useful value, afterwards looper continues to execute but GetConfig returns nil or nothing at all(less than undefined or nil). The logs do not indicate.

In the env directory, an env file is never written to the same directory but to the root directory of the application. Which does not seem to make much of a difference because the balance that's supposed to be rendered in the UI is in fact rendered. So the  voltage api requests appear to be working, and sometimes some of the Kraken api requests are working and then the app fails. 

Note: The chrome profile contains a space "Profile 2" at the tail end of the profile. After escaping the space in the config we seem to have had a better interaction with the Kraken UI but still failing on the same GetConfig function error. 

Happy to hop on a call, and maybe we can get a PR up to resolve this issue since another developer we're in contact with was able to duplicate the situation. 

![image](https://user-images.githubusercontent.com/3687036/203207740-6478a114-4abd-46f0-bd25-a7cfe732780a.png)

**Comment by habibitcoin** [comment]
@JackC27 I think this is resolved in https://github.com/habibitcoin/habibalancer/pull/18/commits/6c88d9d3659be80317b8aa5477690015fbc0cbd1

Want to give the latest development branch a try?

**Comment by JackC27** [comment]
Hello and thank you for that branch. This definitely resolved some of our issues. The nil pointer error is definitely resolved and we confirmed, as updated in the readme, that we are running go 1.19. 
A few new issues have emerged and I'm still waiting on confirmation that the transaction that has processed, was a result of habi, before I'd say this issue is in particular is resolved completely.

 I've noticed that the payload returned from the GetBalanceAndLimits function within strike.go(line 30) is returning a payload and a 200 success message, but then is parsed by a function that is throwing a new error.

![Habi Balence too low nil error](https://user-images.githubusercontent.com/3687036/204056434-26a3098d-c5b1-4589-a2ca-fe19ad407b56.png)
![image](https://user-images.githubusercontent.com/3687036/204056404-1aae2b03-d9b2-4779-a841-53dae0f09590.png)

This prevented Habi from running properly but,
 by commenting out the 3rd if statement(lines 153 - 155) after BTCinUSDlimits,
the application was able to continue running and it may have even successfully completed an arb.
Here is the payload we logged out from the GetBalanceAndLimits call.
`{
    [
        {BTC {BTC 0.00605183}} 
        {USD {USD 0.00}}
    ] 
    [
        {
            BTC { { } { } { } } 
            { { } { } { } }
        } 
        {
            USD {
                {USD 0.00} 
                {USD 3000.00} 
                {USD 3000.00}
            } 
            {
                {USD 0.00} 
                {USD 10000.00} 
                {USD 10000.00}
            }
        }
    ]
}`

**Comment by JackC27** [comment]
We have confirmed that deposits from arb opportunities seem to be properly working.

**Comment by habibitcoin** [comment]
Amazing!

Hmm, couple q's: is StrikeDefaultCurrency set as USD value? I'm thinking an invalid configuration for that may be the culprit of the balance issue + the null quote issue

The typo for rebalance shouldn't matter/impact things

**Comment by JackC27** [comment]
May have spoke too soon with the successful transactions. We had no balance on Deezy's side so...

Other than that, setting the default value to USD did resolve the aforementioned issues...

Next we're closing the already open channel with Deezy and hoping the script will open a new channel for us. 
That worked:
![image](https://user-images.githubusercontent.com/3687036/204065623-1cbfb3ae-4cb7-46bd-bd7b-39d1f734b909.png)

But now we're receiving word that our Kraken balances are too low. I think this may be a config thing. 

We've made a ton of progress though and are calling it a night. We'll keep you posted on any progress tomorrow. 

Thanks again.

**Comment by habibitcoin** [comment]
Can you show the error message you get re; Kraken balance being too low? That should only happen for the withdrawal attempt.

Optimistic either Kraken or Strike will work at this point if the Deezy channel is open and also set ExcludeDeezyFromLiqOps=false if it is your only open channel

**Comment by pjdejoe** [comment]
This is another of the errors we are receiving from Kraken that is
different from the balance being too low.

2022/11/26 18:59:00 Unexpected error performing Kraken withdrawal
2022/11/26 18:59:00 Could not execute request! #7 ([EFunding:Unknown
withdraw key])
2022/11/26 18:59:00 Could not execute request! #7 ([EFunding:Unknown
withdraw key])

**Comment by habibitcoin** [comment]
Ahh good catch, I had hardcoded with Kraken withdrawal address as "umbrel"

You can also now configure this value in the latest commit on the development branch using KrakenWithdrawalAddressKey

**Comment by pjdejoe** [comment]
Thanks all.  It'd be great to know boundaries on amounts.  I'm running into
this error now and not sure where it's coming from:

2022/12/03 21:30:18 Unexpected error performing Kraken withdrawal
2022/12/03 21:30:18 Could not execute request! #7 ([EFunding:Invalid
amount])
2022/12/03 21:30:18 Could not execute request! #7 ([EFunding:Invalid
amount])

**Comment by habibitcoin** [comment]
@pjdejoe the amount is defined in XBT, just needs to be greater than 0.

**Comment by habibitcoin** [comment]
Going to mark this original issue as closed, but pjdejoe feel free to open a new issue if you are still seeing this.
